File: Py (3)/18.py

filas = int(input("Ingrese número de filas: "))
columnas = int(input("Ingrese número de columnas: "))


matriz = []


print("Ingrese los elementos de la matriz:")
for i in range(filas):
    fila = list(map(int, input().split()))
   
    while len(fila) < columnas:
        fila += list(map(int, input().split()))
    matriz.append(fila[:columnas])

print("Matriz original:")
for fila in matriz:
    print(" ".join(map(str, fila)))

f, c = map(int, input("Ingrese fila y columna a modificar: ").split())
nuevoValor = int(input("Ingrese nuevo valor: "))

f -= 1 
c -= 1

if 0 <= f < filas and 0 <= c < columnas:
    matriz[f][c] = nuevoValor
else:
    print("Posición inválida.")

print("Matriz modificada:")
for fila in matriz:
    print(" ".join(map(str, fila)))

File: C++(3)/18.cpp
#include <iostream>
using namespace std;

int main() {
    int filas, columnas;
    cout << "Ingrese número de filas: ";
    cin >> filas;
    cout << "Ingrese número de columnas: ";
    cin >> columnas;

    int** matriz = new int*[filas];
    for (int i = 0; i < filas; i++) {
        matriz[i] = new int[columnas];
    }

    cout << "Ingrese los elementos de la matriz:" << endl;
    for (int i = 0; i < filas; i++) {
        for (int j = 0; j < columnas; j++) {
            cin >> matriz[i][j];
        }
    }

    cout << "Matriz original:" << endl;
    for (int i = 0; i < filas; i++) {
        for (int j = 0; j < columnas; j++) {
            cout << matriz[i][j] << " ";
        }
        cout << endl;
    }

    int f, c, nuevoValor;
    cout << "Ingrese fila y columna a modificar: ";
    cin >> f >> c;
    cout << "Ingrese nuevo valor: ";
    cin >> nuevoValor;

    f--; 
    c--;

    if (f >= 0 && f < filas && c >= 0 && c < columnas) {
        matriz[f][c] = nuevoValor;
    } else {
        cout << "Posición inválida." << endl;
    }

    cout << "Matriz modificada:" << endl;
    for (int i = 0; i < filas; i++) {
        for (int j = 0; j < columnas; j++) {
            cout << matriz[i][j] << " ";
        }
        cout << endl;
    }

    // Liberar memoria
    for (int i = 0; i < filas; i++) {
        delete[] matriz[i];
    }
    delete[] matriz;

    return 0;
}